libraries/AP_AngleSensor/AP_AngleSensor_Params.cpp

```diff
@@ -0,0 +1,54 @@
+#include "AP_AngleSensor_config.h"
+
+#if AP_ANGLESENSOR_ENABLED
+
+#include "AP_AngleSensor.h"
+#include "AP_AngleSensor_Params.h"
+
+
+// table of user settable parameters
+const AP_Param::GroupInfo AP_AngleSensor_Params::var_info[] = {
+    // @Param: TYPE
+    // @DisplayName: Angle Sensor type
+    // @Description: What type of Angle Sensor is connected
+    // @Values: 0:None,1:AS5048B
+    // @User: Standard
+    AP_GROUPINFO_FLAGS("TYPE", 0, AP_AngleSensor_Params, _type, ANGLE_SENSOR_TYPE_NONE, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: BUS
+    // @DisplayName: Angle Sensor Serial Bus Index
+    // @Description: Angle Sensor Serial Bus Index
+    // @Values: 1: I2C1, 2: I2C2, 3: I2C3
+    // @User: Standard
+    AP_GROUPINFO("BUS",    4, AP_AngleSensor_Params, _bus, ANGLE_SENSOR_BUS_DEFAULT),
+
+    // @Param: ADDR
+    // @DisplayName: Serial Bus Address
+    // @Description: Serial Bus Address
+    // @Increment: 1
+    // @User: Standard
+    AP_GROUPINFO("ADDR",    5, AP_AngleSensor_Params, _addr, ANGLE_SENSOR_ADDR_DEFAULT),
```

Comment:
These could be in a i2c sub table that only gets loaded for this driver. As we add more types we don't want all the params all the time. A potentiometer backend would add a pin and voltage and angle pairs.